### PR TITLE
CSS: Warn and fill .css(name, Number)

### LINKED ISSUE #296: Warn about jQuery.cssNumber
See https://github.com/jquery/jquery/issues/4009

Detect assignments to `jQuery.cssNumber` and recommend that it only be done if that property exists; note that it's going away in 4.0.

Recommend that people do not pass raw numbers to `.css()` and instead always pass strings and append `"px"` when appropriate.  Point out `line-height` as a troublesome case.

### PR DESCRIPTION
**Pull request by mgol**
Fixes #296

### CODE REVIEW

**Comment by mgol** [comment]
@dmethvin The first commit is yours, my modifications in the second one.

File: src/css.js [hunk 1]
```diff
@@ -52,3 +82,38 @@ if ( jQueryVersionSince( "3.4.0" ) && typeof Proxy !== "undefined" ) {
...
+
+if ( !jQuery.cssNumber ) {
+	jQuery.cssNumber = {};
+}
+
+// jQuery 3.x uses jQuery.cssNumber internally so we can't warn on access there.
+if ( jQueryVersionSince( "4.0.0" ) ) {
+	migrateWarnProp( jQuery, "cssNumber", jQuery.cssNumber,
+		"jQuery.cssNumber is deprecated" );
+}
```
Comment (mgol) [inline src/css.js:94]: @dmethvin I wonder if this warning is useful even for jQuery 4.x. People may still need conditional code adding more props to that object and then they'd hit the warning even if they guard access with an `if` statement. It seems impossible to warn only on invalid usage. Should we remove it?

File: src/css.js [hunk 2]
```diff
@@ -52,3 +82,38 @@ if ( jQueryVersionSince( "3.4.0" ) && typeof Proxy !== "undefined" ) {
...
+
+function isAutoPx( prop ) {
+
+	// The first test is used to ensure that:
+	// 1. The prop starts with a lowercase letter (as we uppercase it for the second regex).
+	// 2. The prop is not empty.
+	return ralphaStart.test( prop ) &&
+		rautoPx.test( prop[ 0 ].toUpperCase() + prop.slice( 1 ) );
+}
```
Comment (mgol) [inline src/css.js:103]: Copied from https://github.com/jquery/jquery/blob/master/src/css/isAutoPx.js.

File: src/css.js [hunk 3]
```diff
@@ -1,5 +1,35 @@
...
+	//        |         |  / Margin  \  |   | | Right  | |    |
+	//        |---------+-|           |-+---+-| Bottom |-+----|
+	//        |            \ Padding /         \ Left /       |
+	// BEGIN -|                                               |
+	//        |                /---------\                    |
+	//        |               |           |                   |
+	//        |               |  / Min \  |    / Width  \     |
+	//         \--------------+-|       |-+---|          |---/
+	//                           \ Max /       \ Height /
+	rautoPx = /^(?:Border(?:Top|Right|Bottom|Left)?(?:Width|)|(?:Margin|Padding)?(?:Top|Right|Bottom|Left)?|(?:Min|Max)?(?:Width|Height))$/;
```
Comment (mgol) [inline src/css.js:26]: Copied from https://github.com/jquery/jquery/blob/master/src/css/isAutoPx.js.

File: test/css.js [hunk 4]
```diff
@@ -41,3 +41,95 @@ QUnit[ ( jQueryVersionSince( "3.4.0" ) && typeof Proxy !== "undefined" ) ? "test
...
+		"border",
+		"borderWidth",
+		"borderTop",
+		"borderTopWidth",
+		"borderRight",
+		"borderRightWidth",
+		"borderBottom",
+		"borderBottomWidth",
+		"borderLeft",
+		"borderLeftWidth"
```
Comment (mgol) [inline test/css.js:78]: These are all the possible strings that report `true` when passed to `isAutoPx()`.

File: src/css.js (hunk 1 above)
Comment (dmethvin) [reply src/css.js:94]: I agree, let's remove it.

**Comment by mgol** [comment]
@dmethvin I removed warning on `jQuery.cssNumber` & added one comment & test. All in the last commit. PTAL.

**Comment by mgol** [comment]
@dmethvin This is a new warning so we probably need a new minor release, not a patch one. So the next one would be `3.3.0`. Does that sound good? If yes then we need to change the `3.2.1` milestone to `3.3.0`.

**Comment by dmethvin** [comment]
Agreed, I've renamed the next milestone.

**Pull request merged**